src/host/acme_challenge_server.h

```diff
@@ -0,0 +1,283 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
TBH, I like this being on the host side and the extra dependency doesn't bother me. This can all be completely untrusted, so can stay out of the TCB. 

The ACME-endorsed interface currently serves encrypted requests until it's ACME-endorsed, but with the network-endorsed certificate. I'm not sure it's a good idea to add a non-encrypted interface even part time, because users may easily break security guarantees without being aware of it.

Reply:
This being out of TCB is limited to the current SGX situation, and the in-enclave flow benefits from various DoS protection mechanisms, which may be good to have on what must be a public http interface.